[Swift] Verify API docs (#654)
WalletConnect/walletconnect-docs: docs/swift/auth/wallet-usage.md

Pair.configure(metadata: <AppMetadata>)
Auth.configure(account: <Account>, signerFactory: <SignerFactory>)
```

### Subscribe for Authorization Publishers

Following publishers are available to subscribe:

```swift
public var authRequestPublisher: AnyPublisher<(request: AuthRequest, context: VerifyContext?), Never>
public var authResponsePublisher: AnyPublisher<(id: RPCID, result: Result<Cacao, AuthError>), Never>
public let socketConnectionStatusPublisher: AnyPublisher<SocketConnectionStatus, Never>
```

### Authorization Requests

When your `Auth` instance receives requests or responses from a peer client, it will publish a related event. So you should set a subscription to handle them.

```swift
Auth.instance.authRequestPublisher
    .receive(on: DispatchQueue.main)
    .sink { [self self] auth in
        self?.verifyDapp(auth.context)
        self?.showAuthRequest(auth.request)
    }.store(in: &publishers)
```

Auth context provides a domain verification information about `AuthRequest`. It consists of origin of a Dapp from where the request has been sent, validation enum that says whether origin is **unknown**, **valid** or **invalid** and verify url server. 

> To enable verification you have to provide `verifyUrl` in your [AppMetadata](https://docs.walletconnect.com/2.0/ios/core/pairing-usage#pair-configuration). To use a default verify server set this value to `verify.walletconnect.com`. To oup-out just ignore this parameter (`nil` by default).

Reviewer: ```suggestion
To enable verification you have to provide `verifyUrl` in your [AppMetadata](https://docs.walletconnect.com/2.0/ios/core/pairing-usage#pair-configuration). To use the default verify server set this value to `verify.walletconnect.com`. To opt-out just ignore this parameter (`nil` by default).
```

Reviewer: Minor: Is it `nul` or `nil` for opt-out?
